qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -526,6 +526,26 @@ for i in ${image2} ${image4}; do
   remove_image_retry ${CLUSTER2} ${POOL} ${i}
 done
 
+testlog "TEST: request image resync when remote is not primary"
+test_resync_image=test_resync_image
+create_image_and_enable_mirror ${CLUSTER2} ${POOL} ${test_resync_image} ${RBD_MIRROR_MODE}
+write_image ${CLUSTER2} ${POOL} ${test_resync_image} 100
+wait_for_image_replay_stopped ${CLUSTER2} ${POOL} ${test_resync_image}
+wait_for_image_replay_started ${CLUSTER1} ${POOL} ${test_resync_image}
+wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL} ${POOL} ${test_resync_image}
+wait_for_replaying_status_in_pool_dir ${CLUSTER1} ${POOL} ${test_resync_image}
+wait_for_status_in_pool_dir ${CLUSTER2} ${POOL} ${test_resync_image} 'up+stopped'
+stop_mirrors ${CLUSTER1}
+write_image ${CLUSTER2} ${POOL} ${test_resync_image} 100
+demote_image ${CLUSTER2} ${POOL} ${test_resync_image}
+request_resync_image ${CLUSTER1} ${POOL} ${test_resync_image} test_resync_image_id
+start_mirrors ${CLUSTER1}
+wait_for_status_in_pool_dir ${CLUSTER2} ${POOL} ${test_resync_image} 'up+unknown' 'remote image is not primary'
+wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${test_resync_image} 'up+unknown' 'remote image is not primary'
+promote_image ${CLUSTER1} ${POOL} ${test_resync_image}
+compare_images ${CLUSTER1} ${CLUSTER2} ${POOL} ${POOL} ${test_resync_image}
+remove_image_retry ${CLUSTER1} ${POOL} ${test_resync_image}
```

Comment:
> This can be fixed by ensuring that image deletion during a resync is only allowed when the remote image is confirmed to be primary.

If the primary image is deleted during a resync, what guarantees do we have that the secondary gets cleaned up properly and doesn't remain orphaned?

Does this logic also works against deleting a parent image of a clone during resync (especially with deep cloning support) ?

If a primary image is deleted without any resync operation involved, does it automatically trigger deletion of the corresponding secondary image as well? 
make sure it shouldn't affect the existing workflow